### PR TITLE
Add docs about blocking email subaddressing for Google OAuth

### PR DESCRIPTION
**Pull request by mzhong9723**
https://linear.app/clerk/issue/CORE-1300/validate-hd-claim-in-google-oidc-flow-to-prevent-account-take-overs

We are adding a new feature that allows users to block email subaddresses / email aliases for Google OAuth. This new section of the doc will be linked under the dashboard setting to enable this feature.

Screenshot 2024-01-17 at 12 03 51 PM

### CODE REVIEW

**Review by royanger — APPROVED** [review]
Looks good!

**Review by JacobMGEvans — APPROVED** [review]
Looks great!!

**Comment by mzhong9723** [comment]
Will merge this after the following have been deployed to production:
- https://github.com/clerk/dashboard/pull/2175
- https://github.com/clerk/clerk_go/pull/5408

File: docs/authentication/social-connections/google.mdx [hunk 1]
```diff
@@ -50,6 +50,14 @@ In the Clerk Dashboard configuration screen, you should end up with something li
 
 Finally, click **Save** so that the settings are applied. Congratulations! Social connection with Google is now configured for your instance.
 
+### Block email subaddresses
+
+![Block email subaddresses enabled for Google social connection.](/docs/images/authentication-providers/google/google-block-email-subaddresses.webp)
+
+In the Clerk Dashboard configuration screen, you also have an option to block email subaddresses. With this setting on, any Google account with an email address that contains the characters `+`, `=` or `#` will not be able to sign up or be added to existing accounts. By default, this setting will be enabled for new Google social connections.
```
Comment (mzhong9723) [inline docs/authentication/social-connections/google.mdx:57]: >will not be able to sign up or be added to existing accounts

TODO: Probably need to update this as sign ins will be blocked too

**Review by zythosec — APPROVED** [review]

**Review by SokratisVidros — APPROVED** [review]

**Pull request merged**